CMakeLists.txt

```diff
@@ -15,6 +15,23 @@
 cmake_minimum_required(VERSION 3.0.2 FATAL_ERROR)
 # We use the GoogleTest module if it is available (only in CMake 3.9+)
 # It requires CMP0054 and CMP0057 to be enabled.
+
+IF (CMAKE_SYSTEM_NAME MATCHES "Linux")
+  MESSAGE(STATUS "current platform: Linux ")
+ELSEIF (CMAKE_SYSTEM_NAME MATCHES "Windows")
+  MESSAGE(STATUS "current platform: Windows")
+ELSEIF (CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+  MESSAGE(STATUS "current platform: FreeBSD")
+ELSEIF (CMAKE_SYSTEM_NAME MATCHES "Darwin")
+  MESSAGE(STATUS "current platform: MacOS")
+  set(OPENSSL_ROOT_DIR "/usr/local/opt/openssl" )
+  set(OPENSSL_LIBRARIES "/usr/local/opt/openssl/lib" )
+  set(OPENSSL_INCLUDE_DIR "/usr/local/opt/openssl/include" )
```

Comment:
I'm not sold that we should hardcode system openssl in the build.

It may be possible to set these in the make invocation, and to choose system libs using the higher-layer fbcode-builder scripts.

Reply:
We should not. On arm64 macos these path does not exists.

Reply:
I have created https://github.com/facebook/folly/pull/2016 for that.